ilc/client/TransactionManager.js

```diff
@@ -11,18 +11,18 @@ export const slotWillBe = {
 };
 
 export class TransactionManager {
-    #registrySettings;
+    #spinnerConfig;
     #globalSpinner;
     #spinnerTimeout;
 
     #fakeSlots = [];
     #hiddenSlots = [];
     #transactionBlockers = [];
 
-    constructor(registrySettings = {}) {
-        this.#registrySettings = Object.assign({
-            globalSpinner: true
-        }, registrySettings);
+    constructor(registrySettings = {globalSpinner: {enabled: true, customHTML: ''}}) {
+        this.#spinnerConfig = !registrySettings.globalSpinner.customHTML || registrySettings.globalSpinner.enabled === false
```

Comment:
Why do we need to have this difficult statement if we can just save settings?

Reply:
fixed